pkg/cinder/openstack/openstack.go

```diff
@@ -38,15 +40,93 @@ type OpenStack struct {
 	blockstorage *gophercloud.ServiceClient
 }
 
+type Config struct {
+	Global struct {
+		AuthUrl    string `gcfg:"auth-url"`
+		Username   string
+		UserId     string `gcfg:"user-id"`
+		Password   string
+		TenantId   string `gcfg:"tenant-id"`
+		TenantName string `gcfg:"tenant-name"`
+		DomainId   string `gcfg:"domain-id"`
+		DomainName string `gcfg:"domain-name"`
+		Region     string
+	}
+}
+
+func (cfg Config) toAuthOptions() gophercloud.AuthOptions {
+	return gophercloud.AuthOptions{
+		IdentityEndpoint: cfg.Global.AuthUrl,
+		Username:         cfg.Global.Username,
+		UserID:           cfg.Global.UserId,
+		Password:         cfg.Global.Password,
+		TenantID:         cfg.Global.TenantId,
+		TenantName:       cfg.Global.TenantName,
+		DomainID:         cfg.Global.DomainId,
+		DomainName:       cfg.Global.DomainName,
+
+		// Persistent service, so we need to be able to renew tokens.
+		AllowReauth: true,
+	}
+}
+
+func getConfigFromFile(configFilePath string) (gophercloud.AuthOptions, string, error) {
```

Comment:
why is `region` so special in returns?

Reply:
```region``` is not included in ```gophercloud.AuthOptions```, region is included in ```gophercloud.EndpointOpts```, I guess it is better to return ```gophercloud.AuthOptions``` and ```gophercloud.EndpointOpts```, Thanks your suggestion @rootfs